src/ParameterParser.h

```diff
@@ -396,72 +417,110 @@ namespace snowcrash {
-                return false; // Empty string, invalid
+            if (innerSignature.empty()) {
+                return NotParameterType; // Empty string, invalid
             }
 
-            size_t firstSpace = innerSignature.find(" ");
+            if (RegexCapture(innerSignature, "^" PARAMETER_IDENTIFIER "[[:blank:]]*:?", captureGroups) &&
+                !captureGroups[0].empty()) {
 
-            if (firstSpace == std::string::npos) {
-                return RegexMatch(innerSignature, "^" PARAMETER_IDENTIFIER "$");
-            }
+                innerSignature = innerSignature.substr(captureGroups[0].size());
+
+                // If last char is ':', don't strip it from signature
+                if (captureGroups[0].substr(captureGroups[0].size() - 1) == ":") {
+                    innerSignature = ":" + innerSignature;
+                }
 
-            std::string paramName = innerSignature.substr(0, firstSpace);
+                TrimString(innerSignature);
+            }
+            else {
+                return NotParameterType;
+            }
 
-            if (!RegexMatch(paramName, "^" PARAMETER_IDENTIFIER "$")) {
-                return false; // Invalid param name
+            // If contains only parameter name
+            if (innerSignature.empty()) {
+                return OldParameterType;
             }
 
-            // Remove param name
-            innerSignature = innerSignature.substr(firstSpace + 1);
-            size_t descriptionPos = innerSignature.find(snowcrash::DescriptionIdentifier);
+            std::string firstChar = innerSignature.substr(0, 1);
 
-            // Remove description
-            if (descriptionPos != std::string::npos) {
-                innerSignature = innerSignature.substr(0, descriptionPos);
-                innerSignature = TrimString(innerSignature);
+            // If first char is ':' or `=`
+            if (firstChar == ":" || firstChar == "=") {
+
+                innerSignature = innerSignature.substr(1);
+                TrimString(innerSignature);
+
+                if (innerSignature.empty()) {
+                    return NotParameterType; // No sample or default value
+                }
+
+                // Traverse over the value
+                while (!innerSignature.empty()) {
+
+                    std::string first = innerSignature.substr(0, 1);
+
+                    if (first == "`") {
+                        RetrieveEscaped(innerSignature);
+                        TrimString(innerSignature);
+                    }
+                    else if (first == "(") {
+                        break;
+                    }
+                    else {
+                        innerSignature = innerSignature.substr(1);
+                    }
+                }
+
+                return firstChar == ":" ? NewParameterType : OldParameterType;
             }
 
-            size_t attributesPos = innerSignature.find("(");
+            if (innerSignature.substr(0, 1) == "(") {
```

Comment:
Is there some reason to compare one character in known position via substr().
sugest string::operator[]